### PR TITLE
[BUG] Bug in adadelta optimizer's learning rate

### PR DESCRIPTION
**Pull request by ivallesp**
The learning rate is not part of the original AdaDelta algorithm. If we want to add it as a parameter, it should be considered as part of the `update` variable in order to accumulate it into the `delta_accumulators` variable, if not, when it reaches a local minimum it starts bouncing and no total convergence is achieved.

I.e. we should add the actual update in the numerator of the LR adjustment

### CODE REVIEW

**Comment by ivallesp** [comment]
Can somebody help me understand why travis failed? It is pointing out something related to the documentation, but I didn't change anything related to it... Thanks

File: keras/optimizers.py [hunk 1]
```diff
@@ -410,8 +410,8 @@ def get_updates(self, loss, params):
             self.updates.append(K.update(a, new_a))
 
             # use the new accumulator and the *old* delta_accumulator
-            update = g * K.sqrt(d_a + self.epsilon) / K.sqrt(new_a + self.epsilon)
-            new_p = p - lr * update
+            update = lr * g * K.sqrt(d_a + self.epsilon) / K.sqrt(new_a + self.epsilon)
```
Comment (fchollet) [inline keras/optimizers.py]: How is this not exactly equivalent to the previous code?

Comment (ivallesp) [reply keras/optimizers.py]: Look at the line 423, you are accumulating the root mean squared value of the update; if `lr=1` the code is exactly equivalent, but if `lr!=1` the accumulated value in the old version does not take `lr` into account

Comment (ivallesp) [reply keras/optimizers.py]: Look at the line 423, you are accumulating the root mean squared value of the update; if `lr=1` the code is exactly equivalent, but if `lr!=1` the accumulated value in the old version does not take `lr` into account

Comment (ivallesp) [reply keras/optimizers.py]: Am I clear? :-D

**Comment by fchollet** [comment]
I see what you mean.

The current code does not include the learning in the accumulator updates in Adam, Adamax, Nadam, either. The current behavior is consistent across all optimizers, and it is not clear that the proposed change would be an improvement. It would also be a breaking change (it would potentially change convergence behavior), so I think we should avoid it.

**Comment by ivallesp** [comment]
Thanks for the answer. Actually the current implemented Adadelta code diverges when provided with a lr < 1.0, I can provide a simple example in Numpy showing it if it helps :)

**Comment by fchollet** [comment]
@ivallesp: can you check the behavior of the Adadelta implementation in TensorFlow and report your findings? I think the most important thing for us would be consistency with TensorFlow.

**Comment by tanzhenyu** [comment]
@ivallesp , can you elaborate on "if not, when it reaches a local minimum it starts bouncing and no total convergence is achieved"? An example would help.

**Comment by tanzhenyu** [comment]
Also it seems the first tf implementation is from here:
https://github.com/tensorflow/tensorflow/commit/cbfe3a0f5a49b9c0f816a748b2c178afa97bbf16

**Comment by fchollet** [comment]
Closing outdated PR.
